src/devices/machine/z80dma.cpp

```diff
@@ -425,6 +429,11 @@ void z80dma_device::do_search()
 
 void z80dma_device::do_write()
 {
+	if (m_byte_counter && m_dma_mode == dma_mode::SPEC_NEXT)
+	{
+		m_addressB += PORTB_FIXED ? 0 : PORTB_INC ? 1 : -1;
+	}
+
 	switch(TRANSFER_MODE)
```

Comment:
I'm confused: why it needs a pre-increment _before_ the transfer occurs?

Reply:
You need to look at this in combination with block below. Next mode initialized with exact N transfers value, not N+1 as original z80dma